diam/sm/smparser/app.go

```diff
@@ -19,9 +19,34 @@ type Application struct {
 	id                          []uint32 // List of supported application IDs.
 }
 
+// SupportedApp holds properties of each locally supported App
+type SupportedApp struct {
+	ID      uint32
+	AppType string
+	Vendor  uint32
+}
+
+// PrepareSupportedApps prepares a list of locally supported apps
+func PrepareSupportedApps() []*SupportedApp {
+	locallySupportedApps := []*SupportedApp{}
+	for _, app := range dict.Default.Apps() {
+		if app.ID == 0 {
+			continue
+		}
+		addApp := new(SupportedApp)
+		addApp.ID = app.ID
+		addApp.AppType = app.Type
+		for _, vendor := range app.Vendor {
+			addApp.Vendor = vendor.ID
+		}
+		locallySupportedApps = append(locallySupportedApps, addApp)
+	}
+	return locallySupportedApps
+}
+
 // Parse ensures all acct or auth applications in the CE
 // exist in this server's dictionary.
-func (app *Application) Parse(d *dict.Parser) (failedAVP *diam.AVP, err error) {
+func (app *Application) Parse(d *dict.Parser, isServer uint8) (failedAVP *diam.AVP, err error) {
```

Comment:
Another option would be to keep this "role" internal and offer `ParseClient` and `ParseServer` functions.